lib/f_core/net/application/c_tftp_server_tenant.cpp

```diff
@@ -0,0 +1,182 @@
+#include "f_core/net/application/c_tftp_server_tenant.h"
+#include "f_core/os/c_file.h"
+
+#include <cstdio>
+#include <cstring>
+#include <zephyr/fs/fs.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(CTftpServerTenant);
+
+char *inet_ntoa(struct in_addr in)
+{
+    static char buf[INET_ADDRSTRLEN];
+    unsigned char *bytes = (unsigned char *)&in.s_addr;
+
+    snprintf(buf, sizeof(buf), "%d.%d.%d.%d", bytes[0], bytes[1], bytes[2], bytes[3]);
+
+    return buf;
+}
+
+void CTftpServerTenant::Startup() {
+}
+
+void CTftpServerTenant::Cleanup() {
+}
+
+void CTftpServerTenant::Run() {
+    sockaddr srcAddr = {0};
+    socklen_t srcAddrLen = sizeof(srcAddr);
+    uint8_t packet[rwRequestPacketSize] = {0};
+    int rxLen = sock.ReceiveAsynchronous(packet, rwRequestPacketSize, &srcAddr, &srcAddrLen);
+
+    if (rxLen == 0) {
+        return;
+    } else if (rxLen < 0) {
+        LOG_ERR("Failed to receive packet (%d)", rxLen);
+        return;
+    }
+
+    // Opcode is the first two bytes in network order.
+    uint16_t opcode = (packet[0] << 8) | packet[1];
+    switch (opcode) {
+        case RRQ:
+            handleReadRequest(srcAddr, packet, rxLen);
+            break;
+        default:
+            LOG_ERR("Undefined opcode operation %d", opcode);
+    }
+}
+
+int CTftpServerTenant::waitForAck(CUdpSocket &dataSock, const sockaddr &srcAddr, uint16_t blockNum) {
+    static constexpr uint16_t ACK_PKT_SIZE = 4;
+    uint8_t ack[ACK_PKT_SIZE] = {0}; // 2 bytes for opcode, 2 for block number
+    const int maxRetries = 5;
+    int retries = 0;
+
+    while (retries < maxRetries) {
+        int ret = dataSock.ReceiveAsynchronous(ack, 4, const_cast<sockaddr *>(&srcAddr), nullptr);
+        if (ret < 0) {
+            // If no data available yet, wait and retry.
+            if (errno == EWOULDBLOCK || errno == EAGAIN) {
```

Comment:
should we really be blocking for half a second? this is starving other network task tenants 
(kinda a big deal for dep and sensor mod at least where charges should be as low latency as possible

Reply:
So I considered this. I think if we're trying to download flight data, we don't really care about sending Ethernet data to other modules lol

Reply:
Again this was another reason I considered making tftp_server its own thing, which I can do in a new PR. I think if we go that route, Ill also move the receiver code into its own thing under a folder called app/ground

Reply:
oh true I suppose. tho I do think breaking it out is probably good 🙏